### PR TITLE
FIX-#3395: Make `label` param of DMatrix optional

### LINKED ISSUE #3395: Make parameter `label` of Modin DMatrix optional
### Describe the problem

To support drop-in-replacement API for DMatrix

### PR DESCRIPTION
**Pull request by prutskov**
Signed-off-by: Alexey Prutskov <alexey.prutskov@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3395  
- [ ] tests added and passing
- [ ] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/experimental/xgboost/xgboost.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 LOGGER = logging.getLogger("[modin.xgboost]")
 
 
-class DMatrix(xgb.DMatrix):
```
Comment (anmyachev) [inline modin/experimental/xgboost/xgboost.py:28]: @prutskov why don't you inherit anymore?

Comment (prutskov) [reply modin/experimental/xgboost/xgboost.py:28]: Inheritance doesn't include any benefits, because Modin DMatrix provided very limited set of supported properties/methods. Also, inheritance includes additional problems for `doc_checker` for `labels` parameter of `__init__` dunder method.

Comment (anmyachev) [reply modin/experimental/xgboost/xgboost.py:28]: ок

**Review by anmyachev — APPROVED** [review]

**Review by anmyachev — APPROVED** [review]

File: modin/experimental/xgboost/xgboost.py (hunk 1 above)
Comment (anmyachev) [inline modin/experimental/xgboost/xgboost.py:28]: Please make this change in a separate pull request, as it does not apply to the current one.

**Review by anmyachev — CHANGES_REQUESTED** [review]

File: modin/experimental/xgboost/xgboost.py (hunk 1 above)
Comment (prutskov) [reply modin/experimental/xgboost/xgboost.py:28]: If I leave only changes related to fix issue, the new issue will occur - `doc_checker` will found errors for `label` parameter, because in that case docstring from native `xgb.DMatrix` (has differences from our style) will be used. Does it make sense to move inheritance to a separate RR?

Comment (anmyachev) [reply modin/experimental/xgboost/xgboost.py:28]: In this case, `doc_checker` should have found errors before this pull request. How did it work before?

Comment (prutskov) [reply modin/experimental/xgboost/xgboost.py:28]: In my assumption, inheritance of docstrings happens only in case `label=None` (when we aligned with parent class). Before it was positional parameter with own docstring.

Comment (anmyachev) [reply modin/experimental/xgboost/xgboost.py:28]: I think we should fix `doc_checker` in this case as users might rely on inheritance.

Comment (prutskov) [reply modin/experimental/xgboost/xgboost.py]: The root cause of issue is in incorrect docstring of `__init__` method in parent class (`xgb.DMatrix`) for `label` parameter. In our code we skip the check of correctness of inherited docstrings with decorator `_inherit_docstrings`. I don't see the point to add this decorator to the method `__init__`, because we don't require separate docstrings for `__init__` methods in our docstyle. So, I just added `noqa` to skip the check of doc here.

Comment (anmyachev) [reply modin/experimental/xgboost/xgboost.py]: If the documentation does inherit for `__init__`, then it's a problem because it [doesn't match](https://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.DMatrix) the function signature. We need to delete it.

Comment (prutskov) [reply modin/experimental/xgboost/xgboost.py:28]: I added decorator `_remove_docstring`, that removes docstring for object (for `__init__` method in our case).

File: modin/experimental/xgboost/xgboost.py [hunk 2]
```diff
@@ -25,6 +25,23 @@
...
+    Parameters
+    ----------
+    obj : any
+        Object from which need to remove docstring.
+
+    Returns
+    -------
+    `obj` without docstring.
+    """
+    setattr(obj, "__doc__", "")
```
Comment (anmyachev) [inline modin/experimental/xgboost/xgboost.py:41]:
```suggestion
    setattr(obj, "__doc__", None)
```

**Review by anmyachev — DISMISSED** [review]
@vnlitvinov take a look please.

File: modin/experimental/xgboost/xgboost.py (hunk 2 above)
Comment (prutskov) [reply modin/experimental/xgboost/xgboost.py:41]: If we will set `__doc__ to `None`, inherited docstring will be used.

**Review by anmyachev — APPROVED** [review]
